### PR TITLE
PAR-745: Only organisations can edit legal entities on the apply jour…

### PR DESCRIPTION
**Pull request by stephenb01**
…ney.

### CODE REVIEW

File: sync/user.role.par_helpdesk.yml [hunk 1]
```diff
@@ -83,7 +81,6 @@ permissions:
   - 'edit par_data_authority entities'
   - 'edit par_data_enforcement_notice entities'
   - 'edit par_data_inspection_plan entities'
-  - 'edit par_data_legal_entity entities'
```
Comment (jon-edery) [inline sync/user.role.par_helpdesk.yml]: Are we sure we want to remove edit legal entities from the par help desk role?

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsDetailsForm.php [hunk 2]
```diff
@@ -85,18 +86,34 @@ public function buildForm(array $form, FormStateInterface $form_state, ParDataPa
...
+    if ($checkbox === 'partnership_info_agreed_business' && !$par_data_partnership->getBoolean($checkbox)) {
+      // Get current user and check permissions.
+      $account = $this->getCurrentUser();
+      if ($account->hasPermission('edit par_data_legal_entity entities')) {
+        $operations[] = 'edit-entity';
+      }
+      if ($account->hasPermission('add par_data_legal_entity entities')) {
+        $operations[] = 'add';
+      }
+    }
```
Comment (jon-edery) [inline web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsDetailsForm.php]: Are we missing  the following: 
 if ($account->hasPermission('delete par_data_legal_entity entities)) {
       $operations[] = 'delete';
      }

File: sync/user.role.par_helpdesk.yml (hunk 1 above)
Comment (stephenb01) [reply sync/user.role.par_helpdesk.yml]: Yes. The is the message I got from @nao360. The ONLY time they can add/edit legal entities is during the apply stage. After that it is set in stone.

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsDetailsForm.php (hunk 2 above)
Comment (stephenb01) [reply web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsDetailsForm.php]: Yes we should have a delete here, but not sure if delete has been implemented yet or if this is in another jira ticket.

**Comment by kalpaitch** [comment]
@nao360 We need to discuss this requirement. It was requested very urgently yesterday but needs to be thought of much more carefully. It goes against many of the decisions we'd made up to this point, primarily that the business confirming a partnership and a business updating a partnership were the same journey.

**Pull request merged**